### ISSUE TITLE
[Schedule Times] All schedule times being shifted 1 hour out

### ISSUE THREAD

**Issue by astevens93** [netmix/radio-station#237]
**Is your feature request related to a problem? Please describe.**
On the Current Show (and Upcoming Show) widget (and possibly shortcode), have the ability to remove the day in schedule info.

See below:

<img width="169" alt="Screenshot 2020-07-06 at 16 48 35" src="https://user-images.githubusercontent.com/67294458/86612778-8d0be780-bfa8-11ea-8795-6bf92748c651.png">

**Describe the solution you'd like**
Have the option (if you desire to) via the Widget or Shortcode Attribute, to remove the day so it just shows like this:

![image](https://user-images.githubusercontent.com/67294458/86612981-d9efbe00-bfa8-11ea-9549-652d2542c71d.png)

**Describe alternatives you've considered**
It is difficult to hide in CSS as the day is not within it's own container. That was my first port of call. I suspect it could be done through a theme function but that is beyond my personal knowledge.

**Additional context**
Just a feature for a future release. I feel it is a bit strange to state the day in a Current On Air widget. I think this is a historical thing from the days before Netmix.

**Comment by tonyzeoli** [comment]
@astevens93 I'm thinking about your request and my view is your perception is a subjective view of what you want that should be able to be hidden via CSS.

Some people may like it because it reminds the person viewing of what day it is. While for others, like yourself, you believe the person viewing knows what day it is and displaying the day is redundant or inconsequential.

My opinion is to leave it be and allow it to be hidden with CSS.

@majick777 I am not sure how to hide the day here via CSS. Can you instruct?

**Comment by majick777** [comment]
@astevens93 Unfortunately don't see how this is possible with just CSS as there is no wrapper just around the day. 
Since adding one might prove troublesome to time formatting changes in the future, instead I have added some filters to the time format, so for example the day could be omitted from the formatting string. I'll update this ticket when that the next development version is up.

**Comment by astevens93** [comment]
Ok great thanks @majick777 yes a filter sounds like a plan. Much appreciated.

**Comment by majick777** [comment]
@astevens93 I have added time format filters to dev version 2.3.1.0 for the following contexts:
- Master Schedule (Table/Tabs/ List Views)
- Current Show/Upcoming Shows Widgets
- Archive Shortcodes

Here is an example for the Current Show Widget:

```
add_filter( 'radio_station_time_format_start', 'custom_time_format_start', 10, 3 );
function custom_time_format_start( $format, $context, $args ) {
    if ( ( 'current-show' == $context ) && ( 1 == $args['widget'] ) ) {
        $format = 'g:i a'; // 12 hour format without day
    }
    return $format;
}
```

Enjoy!

**Comment by majick777** [comment]
Note to match for shortcode `$args['widget']` will be 0. And to do it for both just exclude that condition entirely.
This would do it for both Current and Upcoming Shows Widgets and Shortcodes:

```
add_filter( 'radio_station_time_format_start', 'custom_time_format_start', 10, 3 );
function custom_time_format_start( $format, $context, $args ) {
    if ( ( 'current-show' == $context ) && ( 'upcoming-shows' == $context ) ) {
        $format = 'g:i a'; 
    }
    return $format;
}
````

**Comment by astevens93** [comment]
Thanks so much for this. Worked a treat in my functions.php file.
However, noticed the latest dev version has pushed my schedules times back by 1 hour. So everything 5-8pm for example, becomes 4-7pm and so on and so forth.

**Comment by majick777** [comment]
@astevens93 The entire schedule? That does seem strange. Can you let me know your timezone settings and which dev versions and I can attempt to reproduce.

**Comment by astevens93** [comment]
Hi @majick777 sorry it's taken me some time to get back to you.
So I've just double checked this by disabling dev version 2.3.1.9 and enabling 2.3.1.10 to confirm I'm correct in what I say, and unfortunately I am. 

When switching to 2.3.1.10 it shifts all the scheduled show times back by 1 hour. So 12:00am - 02:00 am becomes 11:00pm - 01:00am and so forth for every single show.

See below timezone settings. As I say, it only happens when i enabled 2.3.1.10 - very strange indeed.

![image](https://user-images.githubusercontent.com/67294458/87181579-77fdc400-c2da-11ea-9be2-ef7b6a9f7ebb.png)

**Comment by majick777** [comment]
@astevens93 Okay thanks for that will test it out, can't have that happening!

**Comment by majick777** [comment]
@astevens93 Found the issue. Nothing to do with schedule calculation, just the final display output. Should be all good now in 2.3.1.11

**Comment by astevens93** [comment]
Thanks yes that sorted the time shift issue!  I have noticed that on tabbed schedule Version 2.3.1.11 doesn't default to the current day tab, it just loads on Monday regardless of the day. For example today is Saturday but the tabbed view loads on Monday.

**Comment by majick777** [comment]
Resolved in 2.3.1.12 as per #239
